pypyr/dsl.py

```diff
@@ -332,13 +391,25 @@ def run_conditional_decorators(self, context):
                                                         self.invoke_step)
                     else:
                         self.invoke_step(context=context)
-                except Exception as ex_info:
+                except Exception as exc_info:
+                    self.save_error(context=context, exception=exc_info)
                     if swallow_me:
                         logger.error(
```

Comment:
should we use `logger.warning` here instead `error`? seems swallowed exception are recoverable

Reply:
No, `error` is good. . . although recoverable, it's still an error. The ultimate pypyr exit code will still be 0 assuming the err recovered and the pipeline finished successfully.

Reply:
it is a bit confusing for me, when to use `warning` or `error` especially in cases like that,  I'm try to follow to the suggestions at https://stackoverflow.com/a/2031209/1027110. In case above swallowed exception is fatal for operation which is try, so we should use `error` log level here, but on the other hand at my pipelines I am always using `swallow` only when I can safely recover the exception by the next step or it is not crucial for the whole pipeline.

Reply:
to be honest, I'm not particularly _scientific_ about it. . . 

a pipeline operator could well be using `swallow` to catch a hard error which a next step will compensate for by doing something different because of the error (which is pretty much what you're describing with your use case). . . in all cases, we're talking about an "error". 

this being a pipeline, the important thing to the calling process is the exit code - if that's !=0, the raised errors start being meaningful. if exit code == 0, any raised errors are still errors that the pipeline author decided is not important to the success of the pipeline. 

In fact, slightly changing the topic, I have been thinking that the difference between INFO and DEBUG in pypyr is too narrow: INFO pumps out a lot of info already, and DEBUG is well, everything and then some and not for the casual user. There should prob be a level above INFO which just shows the step `description` or `running step {name}`... #132 But that's a topic for another day. . .

Reply:
Great, thanks for the explanation. #132 is also nice to have for me. Probably I'm trying to deal with my own problem. I have [sentry](https://sentry.io) connected to my app, but it does not know anything about `swallowing`. So what do I want to do is to stop to report such kind of exceptions somehow. 

Also is it useful to know at the pipeline level was exception `swallow`'ed or not? should I add `swallow` argument to the context 'runErrors'?

Reply:
ooh, that's a great idea, thank you! How about adding `swallowed`, make it clear it's bool?

Reply:
Done!